### ISSUE TITLE
kernel-kit: generated source code fails to compile ndiswrapper

### ISSUE THREAD

**Issue by rizalmart** [puppylinux-woof-CE/woof-CE#1862]
I  noticed that the kernel source code and kernel headers produced by kernel-kit failed to compile ndiswrapper due to missing makefiles and header files which was present on the vanilla kernel source code.

I suggest that avoid deletions in "include" folder and "script" folder of linux source code.

**Comment by peabee** [comment]
Is it possible to identify which files are not provided and where in kernel-kit/build.sh they are being deleted??

**Comment by rizalmart** [comment]
@peabee 
Try to load the generated linux kernel source sfs and linux header sfs. Don't forget to load the devx module.

Then get the latest ndiswrapper source code and compile it on terminal.

I noticed that some makefiles and some headers are missing which ndiswrapper failed to compile. The ndiswrapper source code depends on "/usr/src/linux/scripts" and "/usr/src/linux/includes". It means there are files that got deleted on this two mentioned folders

I tested it using lxpupsc kernel source 5.7.x

**Comment by gyrog** [comment]
On ScPup64, "/usr/src/linux/scripts" is in the "kernel_sources" sfs,
and there is a "/usr/src/linux/include" in the "kernel_sources" sfs.

**Comment by 01micko** [comment]
I just built ndiswrapper in a very recent build of 5.4.61. Only problem is multiple versions of `version.h`. I recall we made a symlink ages ago because it was either nvidia or amd proprietary driver complained.

Anyway, patching ndiswrapper source is trivial.

```patch
diff -rup ndiswrapper-1.63/driver/Makefile ndiswrapper-1.63-patched/driver/Makefile
--- ndiswrapper-1.63/driver/Makefile	2020-05-03 22:18:33.000000000 +1000
+++ ndiswrapper-1.63-patched/driver/Makefile	2020-09-12 07:53:02.759129684 +1000
@@ -29,9 +29,9 @@ $(error No .config found in $(KBUILD), p
 endif
 
 ifneq (,$(wildcard $(KBUILD)/include/linux/version.h))
-ifneq (,$(wildcard $(KBUILD)/include/generated/uapi/linux/version.h))
-$(error Multiple copies of version.h found, please clean your build tree)
-endif
+#ifneq (,$(wildcard $(KBUILD)/include/generated/uapi/linux/version.h))
+#$(error Multiple copies of version.h found, please clean your build tree)
+#endif
 endif
 
 # Kernel Makefile doesn't always know the exact kernel version, so we
```
